File: python/Singleton/complete/SingletonLazy.py
class SingletonLazy:
    obj = None
    
    def __init__(self):
        self.log = ""
            
    @staticmethod
    def get_instance():
        if(SingletonLazy.obj == None):
            SingletonLazy.obj = SingletonLazy()
        return SingletonLazy.obj
    
    def add_to_log(self, string):
        if(self.log == ""):
            self.log += string
        else:
            self.log += "\n"+string
        
    def get_log(self):
        return self.log

File: java/Singleton/complete/SingletonLazy.java

public class SingletonLazy{
  public static SingletonLazy obj = null;
  private String log;

  private SingletonLazy(){
    log = "";
  }

  public static SingletonLazy get_instance(){
    //create singleton object if it has not been created, then return the object
    if(obj == null){
      obj = new SingletonLazy();
    }

    return obj;
  }

  public void add_to_log(String s){
    //add string s to the current log
    if(log.equals(""))
      log = s;
    else
      log = log + "\n" + s;
  }

  public String get_log(){
    //return the current log
    return log;
  }

}
